x/profiles/keeper/msgs_server_profile.go

```diff
@@ -45,7 +45,7 @@ func (k msgServer) SaveProfile(goCtx context.Context, msg *types.MsgSaveProfile)
 
 	// If the DTag changes, delete all the previous DTag transfer requests
 	if profile.DTag != msg.DTag {
-		k.DeleteAllDTagTransferRequests(ctx, msg.Creator)
+		k.DeleteAllUserIncomingDTagTransferRequests(ctx, msg.Creator)
 	}
```

Comment:
I think this part should be moved the last part of the function to ensure that no error happens after this part.

Reply:
Thanks for the tip. I actually ended up removing this entirely and moving it inside the `StoreProfile` method instead. This should make sure that if such method is called from anywhere else, the DTag requests will be deleted properly.